addons/payment_nuvei/models/payment_transaction.py

```diff
@@ -144,9 +144,20 @@ def _compare_notification_data(self, notification_data):
         if self.provider_code != 'nuvei':
             return super()._compare_notification_data(notification_data)
 
+        # When a user declines to pay and leaves the payment page, no information
+        # is sent back to odoo via the endpoint. As such there is no currency or
+        # amount set so we return early. This only occurs in the leaving flow so
+        # no issue should arise leaving early.
+        if not notification_data:
+            return
```

Comment:
This works, but we could rather immediately redirect the user to /payment/status in the return controller instead of trying to process empty payment data. This would also allow us to stop building "fake" payment data just to match the transaction.

Reply:
As discussed, Nuvei cancels transactions without payment data during processing, so it's better for the user experience to go through the whole processing and see a cancelled payment than to have to click "skip" on a draft payment warning.